client/src/main/java/tds/session/Session.java

```diff
@@ -42,4 +53,129 @@ public int getType() {
     public void setType(int type) {
         this.type = type;
     }
+
+    /**
+     * @return The time when the {@link Session} was started.
+     * <p>
+     *     Should handle date and time in UTC
+     * </p>
+     */
+    public Timestamp getDateBegin() {
+        return dateBegin;
+    }
+
+    public void setDateBegin(Timestamp dateBegin) {
+        this.dateBegin = dateBegin;
+    }
+
+    /**
+     * @return The time when the {@link Session} was ended.
+     * <p>
+     *     Should handle date and time in UTC
+     * </p>
+     */
+    public Timestamp getDateEnd() {
+        return dateEnd;
+    }
+
+    public void setDateEnd(Timestamp dateEnd) {
+        this.dateEnd = dateEnd;
+    }
+
+    /**
+     * @return The date and time when a {@link Session} was changed/updated.
+     * <p>
+     *     Should handle date and time in UTC.  Currently, this field is only updated when the {@link Session}'s status
+     *     changes.
+     * </p>
+     */
+    public Timestamp getDateChanged() {
+        return dateChanged;
+    }
+
+    public void setDateChanged(Timestamp dateChanged) {
+        this.dateChanged = dateChanged;
+    }
+
+    /**
+     * @return The time when the {@link Session} was "visited".
+     * <p>
+     *     Should handle date and time in UTC.  This field is related to evaluating the checkin time.
+     * </p>
+     */
+    public Timestamp getDateVisited() {
+        return dateVisited;
+    }
+
+    public void setDateVisited(Timestamp dateVisited) {
+        this.dateVisited = dateVisited;
+    }
+
+    /**
+     * @return The client name for the current tenant
+     * <p>
+     *     Correlates to (at least) the session.externs view, the sessions._externs table and the configs.client_externs
+     *     table.
+     * </p>
+     */
+    public String getClientName() {
+        return clientName;
+    }
+
+    public void setClientName(String clientName) {
+        this.clientName = clientName;
+    }
+
+    /**
+     * @return The identifier of the Proctor managing this {@link Session}.
+     */
+    public Integer getProctorId() {
+        return proctorId;
+    }
+
+    public void setProctorId(Integer proctorId) {
+        this.proctorId = proctorId;
+    }
+
+    /**
+     * @return The identifier of the "browser" (the workstation?) managing this {@link Session}.
+     */
+    public UUID getBrowserKey() {
+        return browserKey;
+    }
+
+    public void setBrowserKey(UUID browserKey) {
+        this.browserKey = browserKey;
+    }
+
+    /**
+     * Logic from line 1103 of StudentDLL._ValidateTesteeAccessProc_SP:
```

Comment:
I understand why you've added references to the legacy code, and if we want to do this we should mark them with TODO - DELETE or something like that.  I would like all new code to document what the method represents and not refer to old code since the old code will eventually be deleted.

Secondly, I'm wondering if this logic should reside in the service/data layer rather than putting logic into the POJOs.